README.md

```diff
@@ -15,14 +15,14 @@ In cases where more dynamic customization is sensible, suitable environment vari
 
 Any option without a default is required.
 
-| Option                | Definition                                                          | Type           | Default             | Supports dynamic modification |
-|-----------------------|---------------------------------------------------------------------|----------------|---------------------|-------------------------------|
-| `build-backend`       | The wrapped build backend (e.g. `setuptools.build_meta`)            | string         |                     | N                             |
-| `commit-file`         | The file in which to write the git commit hash                      | string         | "" (No file)        | N                             |
-| `dependencies-file`   | The path to the `dependencies.yaml` file to use                     | string         | "dependencies.yaml" | Y                             |
-| `matrix-entry`        | A `;`-separated list of `=`-delimited key/value pairs               | string         | ""                  | Y                             |
-| `require-cuda`        | If false, builds will succeed even if nvcc is not available         | bool           | true                | Y                             |
-| `requires`            | List of build requirements (in addition to `build-system.requires`) | list[str]      | []                  | N                             |
+| Option                | Definition                                                          | Type           | Default                     | Supports dynamic modification |
+|-----------------------|---------------------------------------------------------------------|----------------|-----------------------------|-------------------------------|
+| `build-backend`       | The wrapped build backend (e.g. `setuptools.build_meta`)            | string         |                             | N                             |
+| `commit-file`         | The file in which to write the git commit hash                      | string         | "<project_name>/GIT_COMMIT" | N                             |
```

Comment:
Probably OK for now, but one consequence of this changes it that it is no longer for a project to state that it does not want a commit file embedded. If we change the behavior in this way is there any way we could reenable that in the future without a breaking change? Just trying to future-proof the design.

Reply:
See my discussion at https://github.com/rapidsai/rapids-build-backend/pull/30#discussion_r1595922544. I had initially thought that I could pass a `null` value to signify this, but it seems that toml doesn't support `null`.

If we want to allow disabling it entirely then we have a tristate: off, auto, and on (with a specified path.) I'll think about ways to express this.

Reply:
What if we allowed zero-or-more commit files? So the following are all valid:

```toml
commit-files = [
    "package/GIT_COMMIT",
]
```

```toml
commit-files = [
    "package_a/GIT_COMMIT",
    "package_b/GIT_COMMIT",
]
```

```toml
commit-files = []  # No files are written
```

And if `commit-files` isn't specified then it does the automatic behavior that we normally want.

Reply:
oh that's a really nice way to solve this. And then if `[]` is passed, `rapids-build-backend` doesn't even both trying to determine the commit?

I support that!

Reply:
I've implemented this idea.

Reply:
Sorry I'm slow in responding, but yes that seems fine to me.